selfdrive/car/hyundai/carstate.py

```diff
@@ -0,0 +1,220 @@
+    ("ACC_REQ", "TCS13", 0),
+    ("DriverBraking", "TCS13", 0),
+    ("DriverOverride", "TCS13", 0),
+
+    ("ESC_Off_Step", "TCS15", 0),
+
+    ("CF_Lvr_GearInf", "LVR11", 0),        #Transmission Gear (0 = N or P, 1-8 = Fwd, 14 = Rev)
+
+    ("CR_Mdps_DrvTq", "MDPS11", 0),
+
+    ("CR_Mdps_StrColTq", "MDPS12", 0),
+    ("CF_Mdps_ToiActive", "MDPS12", 0),
+    ("CF_Mdps_ToiUnavail", "MDPS12", 0),
+    ("CF_Mdps_FailStat", "MDPS12", 0),
+    ("CR_Mdps_OutTq", "MDPS12", 0),
+
+    ("VSetDis", "SCC11", 0),
+    ("SCCInfoDisplay", "SCC11", 0),
+    ("ACCMode", "SCC12", 1),
+
+    ("SAS_Angle", "SAS11", 0),
+    ("SAS_Speed", "SAS11", 0),
+  ]
+
+  checks = [
+    # address, frequency
+    ("MDPS12", 50),
+    ("MDPS11", 100),
+    ("TCS15", 10),
+    ("TCS13", 50),
+    ("CLU11", 50),
+    ("ESP12", 100),
+    ("EMS12", 100),
+    ("CGW1", 10),
+    ("CGW4", 5),
+    ("WHL_SPD11", 50),
+    ("SCC11", 50),
+    ("SCC12", 50),
+    ("SAS11", 100)
+  ]
+
+  return CANParser(DBC[CP.carFingerprint]['pt'], signals, checks, 0)
+
+def get_camera_parser(CP):
+
+  signals = [
+    # sig_name, sig_address, default
+    ("CF_Lkas_LdwsSysState", "LKAS11", 0),
+    ("CF_Lkas_SysWarning", "LKAS11", 0),
+    ("CF_Lkas_LdwsLHWarning", "LKAS11", 0),
+    ("CF_Lkas_LdwsRHWarning", "LKAS11", 0),
+    ("CF_Lkas_HbaLamp", "LKAS11", 0),
+    ("CF_Lkas_FcwBasReq", "LKAS11", 0),
+    ("CF_Lkas_ToiFlt", "LKAS11", 0),
+    ("CF_Lkas_HbaSysState", "LKAS11", 0),
+    ("CF_Lkas_FcwOpt", "LKAS11", 0),
+    ("CF_Lkas_HbaOpt", "LKAS11", 0),
+    ("CF_Lkas_FcwSysState", "LKAS11", 0),
+    ("CF_Lkas_FcwCollisionWarning", "LKAS11", 0),
+    ("CF_Lkas_FusionState", "LKAS11", 0),
+    ("CF_Lkas_FcwOpt_USM", "LKAS11", 0),
+    ("CF_Lkas_LdwsOpt_USM", "LKAS11", 0)
+  ]
+
+  checks = []
+
+  return CANParser(DBC[CP.carFingerprint]['pt'], signals, checks, 2)
+
+class CarState(object):
+  def __init__(self, CP):
+
+    self.CP = CP
+
+    # initialize can parser
+    self.car_fingerprint = CP.carFingerprint
+
+    # vEgo kalman filter
+    dt = 0.01
+    # Q = np.matrix([[10.0, 0.0], [0.0, 100.0]])
+    # R = 1e3
+    self.v_ego_kf = KF1D(x0=np.matrix([[0.0], [0.0]]),
+                         A=np.matrix([[1.0, dt], [0.0, 1.0]]),
+                         C=np.matrix([1.0, 0.0]),
+                         K=np.matrix([[0.12287673], [0.29666309]]))
+    self.v_ego = 0.0
+    self.left_blinker_on = 0
+    self.left_blinker_flash = 0
+    self.right_blinker_on = 0
+    self.right_blinker_flash = 0
+
+  def update(self, cp, cp_cam):
+    # copy can_valid
+    self.can_valid = cp.can_valid
+
+    # update prevs, update must run once per Loop
+    self.prev_left_blinker_on = self.left_blinker_on
+    self.prev_right_blinker_on = self.right_blinker_on
+
+    self.door_all_closed = True
+    self.seatbelt = cp.vl["CGW1"]['CF_Gway_DrvSeatBeltSw']
+
+    self.brake_pressed = cp.vl["TCS13"]['DriverBraking']
+    self.esp_disabled = cp.vl["TCS15"]['ESC_Off_Step']
+
+    self.park_brake = False
```

Comment:
["CGW1"]['CF_Gway_ParkBrakeSw']

EPB11 is dedicated to the park brake, also an option

Reply:
yeah, I can fill this too.